tests/io/dask/dataframe/test_repartition.py

```diff
@@ -0,0 +1,114 @@
+    """
+    dfs = [df]
+    store_dataframes_as_dataset__iter(
+        dfs,
+        store=dataset_factory_function_scope.store_factory,
+        dataset_uuid=dataset_factory_function_scope.dataset_uuid,
+        partition_on=["part"],
+    )
+
+    repartition_ratio = calculate_repartition_ratio(
+        dataset_factory=dataset_factory_function_scope,
+        table_name=SINGLE_TABLE,
+        dataset_metadata_frac=frac,
+        target_nrows_per_partition=target_nrows_per_partition,
+    ).compute()
+
+    ddf = read_dataset_as_ddf(factory=dataset_factory_function_scope)
+    # call optimize here and find the ddf length
+    ddf_before_repartition = dask.optimize(
+        ddf, optimizations=[dask.dataframe.optimize]
+    )[0]
+
+    ddf = repartition_ddf(ddf, repartition_ratio=repartition_ratio)
+    # length of ddf after repartition should be less than before, to test the block fusing
+    # but if repartition_ratio==1, then the length should remain same.
+    assert len(ddf.dask.keys()) <= len(ddf_before_repartition.dask.keys())
+
+    assert (ddf.map_partitions(len).compute() == target_nrows_per_partition).all()
+
+
+@pytest.mark.parametrize("frac", (0.3, 0.1))
+@pytest.mark.parametrize(
+    "df,target_nrows_per_partition",
+    (
+        # repartition by decreasing partition size, increasing number of partitions
+        (pd.DataFrame({"A": range(100), "part": ["A", "B", "C", "D"] * 25}), 97),
+        (pd.DataFrame({"A": range(100), "part": ["A", "B", "C", "D"] * 25}), 31),
+        # repartition by increasing partition size, decreasing number of partitions
+        (pd.DataFrame({"A": range(2 ** 6), "part": list("0123456789abcdef") * 4}), 11),
+        (pd.DataFrame({"A": range(10), "part": range(10)}), 7),
+        (pd.DataFrame({"A": range(10), "part": range(10)}), 1),
+    ),
+)
+def test_calculate_repartition_ratio_approximate_match(
+    dataset_factory_function_scope, df, target_nrows_per_partition, frac
+):
+    dfs = [df]
+    store_dataframes_as_dataset__iter(
+        dfs,
+        store=dataset_factory_function_scope.store_factory,
+        dataset_uuid=dataset_factory_function_scope.dataset_uuid,
+        partition_on=["part"],
+    )
+
+    repartition_ratio = calculate_repartition_ratio(
+        dataset_factory=dataset_factory_function_scope,
+        table_name=SINGLE_TABLE,
+        dataset_metadata_frac=frac,
+        target_nrows_per_partition=target_nrows_per_partition,
+    ).compute()
+
+    ddf = read_dataset_as_ddf(factory=dataset_factory_function_scope)
+    # call optimize here and find the ddf length
+    ddf_before_repartition = dask.optimize(
+        ddf, optimizations=[dask.dataframe.optimize]
+    )[0]
+    # import IPython
+    # IPython.embed()
+    ddf = repartition_ddf(ddf, repartition_ratio=repartition_ratio)
+    # find length after repartition
+    # length of ddf after repartition should be less than before.
+    assert len(ddf.dask.keys()) <= len(ddf_before_repartition.dask.keys())
```

Comment:
Can you please add this as a dedicated test? This is quite an important feature of this repartitioning and should not be squeezed in as a simple assert. Writing a dedicated test with explicit parameters for the use case we're about to test is much easier to debug. Every unit test should test one feature not multiple

Reply:
This assert can only be true if the repartition ratio is indeed `>=1` and the fusing works. That the repartition ratio is in the area we expect it to be is not really clear to me at this stage of the test

Reply:
Have written dedicated test case, with different repartition ratio values-  greater,  less than, and equal to 1.